packages/xcoobee-cookie-kit-react/src/CookieKitContainer.js

```diff
@@ -159,66 +167,72 @@ export default class CookieKitContainer extends React.PureComponent {
     }
   }
 
-  componentDidMount() {
-    if (this.state.countryCode) {
-      this.getCookieConsents();
-    } else {
-      CookieConsentsManager.fetchCountryCode()
-        .catch((error) => {
-          // console.log("CookieKitContainer#componentDidMount#fetchCountryCode#catch");
-          console.error(error);
-          return null;
-        })
-        .then((countryCode) => {
-          saveCountryCode(countryCode);
-          this.setState({ countryCode });
-          this.getCookieConsents();
-        })
-        .catch(handleErrors);
+  onLoginStatusChange(loginStatus) {
+    // console.log("CookieKitContainer#onLoginStatusChange");
+    const { isLoginStatusChecked } = this.state;
+
+    this.setState({ loginStatus });
+
+    if (!isLoginStatusChecked) {
+      this.getCountryCode();
+      this.setState({ isLoginStatusChecked: true });
     }
   }
 
-  // componentDidUpdate(prevProps, prevState) {
-  //   console.log("CookieKitContainer#componentDidUpdate");
-  //   if (this.props !== prevProps) {
-  //     console.log("props changed:");
-  //     console.dir(this.props);
-  //   }
-  //   if (this.state !== prevState) {
-  //     console.log("state changed:");
-  //     console.dir(this.state);
-  //   }
-  // }
-
   getCookieConsents() {
+    // console.log("CookieKitContainer#getCookieConsents");
     const cachedCookieConsents = cookieConsentsCache.get();
+
     if (cachedCookieConsents) {
       // console.log("Using cached cookie consents!");
       this.setCookieConsents("cached", cachedCookieConsents);
+      this.setState({ isConsentCached: true });
+
       return;
     }
 
     const { campaignReference } = this.props;
+    const { loginStatus } = this.state;
 
     const isConnected = !!campaignReference;
-    const { accessToken } = this.state;
 
-    if (isConnected && accessToken) {
-      this.resolveConnectedCookieConsents().catch(handleErrors);
-    } else {
+    if (!isConnected || !loginStatus) {
       this.fallBackToHostDefaults();
     }
   }
 
+  getCountryCode() {
```

Comment:
I think it will make the code more clear if it doesn't call `this.getCookieConsents()` once the country code is known. Instead, can be always return a Promise from this function? Then where it is called in `onLoginStatusChange`, we can call `this.getCookieConsents`?

I added a few code suggestions related to this.

Reply:
Please see new comments related to this up above in the `onLoginStatusChange` method.